tests/test_integration/test_dace_parsing.py

```diff
@@ -94,6 +98,10 @@ def call_stencil_object(locoutp, locinp):
         defn(locoutp, par=locinp)
 
     call_stencil_object(locoutp=outp, locinp=inp)
+
+    if cp is not None:
+        cp.cuda.Device(0).synchronize()
+
     outp.device_to_host(force=True)
```

Comment:
Is this device synchronization required for other backends as well, or is it because `dace.program` doesn't synchronize as bare stencils do?

Reply:
In general, we rely on the device_to_host copy to be synchronous with previous CUDA calls. This is the case if everything is running on the same CUDA stream. Since that seemed to not work, this was an attempt to solve the issue by explicitly putting a device-wide barrier. before copying to host memory.